packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1498,17 +1498,20 @@ export const renderToHTMLOrFlight: AppPageRender = (
   query,
   renderOpts
 ) => {
-  // TODO: this includes query string, should it?
-  const pathname = validateURL(req.url)
+  if (!req.url) {
+    throw new Error('Invalid URL')
```

Comment:
Why we don't need to revalidate the req.url anymore?

Reply:
The parse function below performs the validation!